cordova-lib/src/cordova/compile.js

```diff
@@ -40,7 +40,7 @@ module.exports = function compile(options) {
     }).then(function() {
         return hooksRunner.fire('after_compile', options);
     }, function(error) {
-        events.emit('log', 'ERROR building one of the platforms: ' + error + '\nYou may not have the required environment or OS to build this project');
+        events.emit('warn', 'ERROR building one of the platforms: ' + error + '\nYou may not have the required environment or OS to build this project');
```

Comment:
I'm not sure if we should continue emitting message here at all, because it is not consistent with other similar places, e.g. but `build` and `emulate` do not emit this message.  Also the message is not very informative, as `compile` might fail due to a lot of different reasons.

In general for cases like this, when initial error message might not be very descriptive, i'd rather reject a promise with (or throw in case of sync code) a new `CordovaError` with appropriate message and original error object, attached.